aten/src/ATen/native/cpu/GridSamplerKernel.cpp

```diff
@@ -620,23 +639,17 @@ struct ApplyGridSample<scalar_t, 2, GridSamplerInterpolation::Bilinear,
     integer_t i_gInp_sw_offset_arr[iVec::size()];
     // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
     integer_t i_gInp_se_offset_arr[iVec::size()];
-    i_gInp_nw_offset.store(i_gInp_nw_offset_arr);
-    i_gInp_ne_offset.store(i_gInp_ne_offset_arr);
-    i_gInp_sw_offset.store(i_gInp_sw_offset_arr);
-    i_gInp_se_offset.store(i_gInp_se_offset_arr);
-
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_nw_mask_arr[iVec::size()];
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_ne_mask_arr[iVec::size()];
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_sw_mask_arr[iVec::size()];
-    // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-    integer_t i_se_mask_arr[iVec::size()];
-    nw_mask.store(i_nw_mask_arr);
-    ne_mask.store(i_ne_mask_arr);
-    sw_mask.store(i_sw_mask_arr);
-    se_mask.store(i_se_mask_arr);
+    if (input_requires_grad) {
+      auto i_gInp_nw_offset = i_y_n * iVec(inp_W) + i_x_w;
```

Comment:
This change and the one below come from @to-mi fix for the 2D case, which I haven't looked into. So I don't really understand what happens here, but I could look into it if needed

Reply:
Yes, this one look like you create a variable that ends up not being used.

Reply:
The store method of `i_gInp_nw_offset` variable looks like it affects the value `i_gInp_nw_offset_arr` so even though `i_gInp_nw_offset` is not used afterwards, it seems to still be useful ?
For the `i_gInp_nw_offset_arr` variable, it is only used if input_requires_grad is True but I still need to declare it.
I could use the lambda trick for the four such variables but it would be quite heavy I think

Reply:
For small objects you don't have to for sure. Tensor is a bit expensive, that's why I think it is worth doing.

And indeed the store actually stores in the arr! Good to know! This looks good then.